### PR TITLE
197 as a dev i want to refactor the sidemenu component

### PR DESCRIPTION
**Pull request by SamuelPelletierEvraire**
- [x]  First button return to the HomePage.
- [x]  Second button navigate to SearchPage.
- [x]  Should not be displayed unless the hamburger button is clicked.
- [x]  Should collapse if the user clicks outside the side menu area.
- [x]  No animation should be present when opening and closing.
- [ ]  Button at the bottom report an issue on our GitHub. https://github.com/ai-cfia/fertiscan-frontend/issues/282

### CODE REVIEW

**Comment by ChromaticPanic** [comment]
Might as well update this to use tailwind

**Comment by SamuelPelletierEvraire** [comment]
> Might as well update this to use tailwind
@ChromaticPanic 
Not sure to understand this is already done

**Comment by SamuelPelletierEvraire** [comment]
All sx is done in another pr see: https://github.com/ai-cfia/fertiscan-frontend/pull/318

File: src/app/layout.tsx [hunk 1]
```diff
@@ -13,14 +13,18 @@ export default function RootLayout({
...
+  };
+
   return (
     <html lang="en">
       <body>
         <AppRouterCacheProvider>
           <ThemeProvider theme={theme}>
-            <SideNav open={sideNavOpen} />
+            <SideNav open={sideNavOpen} onClose={handleDrawerClose} />
             {/* Margin adjustment based on SideNav state */}
```
Comment (k-allagbe) [inline src/app/layout.tsx:26]: Please remove this comment.

File: src/app/layout.tsx [hunk 2]
```diff
@@ -13,14 +13,18 @@ export default function RootLayout({
...
-            <Box marginLeft={sideNavOpen ? "240px" : 0}>
+            <Box>
```
Comment (k-allagbe) [inline src/app/layout.tsx:27]: This box might be unnecessary.

File: src/app/layout.tsx (hunk 1 above)
Comment (SamuelPelletierEvraire) [reply src/app/layout.tsx]: done

**Review by k-allagbe — APPROVED** [review]

**Pull request merged**